code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -141,22 +193,99 @@
 
 				user.visible_message("\red [user] removes the power cell from [A]!", "You remove the power cell.")
 
-//TODO: Matter decompiler.
+
+//NEW MATTER DECOMPILER
 /obj/item/weapon/matter_decompiler
 
 	name = "matter decompiler"
 	desc = "Eating trash, bits of glass, or other debris will replenish your stores."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "decompiler"
-
-	//Metal, glass, wood, plastic.
+	//Temporary storage to assess amount gained per use
+	var/list/store_buffer = list(
+		"metal" = 0,
+		"glass" = 0,
+		"wood" = 0,
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
+		)
+	//Metal, glass, wood, plastic, plasteel, biomass
 	var/list/stored_comms = list(
 		"metal" = 0,
 		"glass" = 0,
 		"wood" = 0,
-		"plastic" = 0
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
 		)
 
+/obj/item/weapon/matter_decompiler/verb/check_salvage()
+
+	set name = "Check Salvage Levels"
+	set desc = "Review levels of stored salvage."
+	set category = "Robot Commands"
+	var/S = null
+	var/amount = null
+
+	for(S in src.stored_comms)
+		amount = src.stored_comms[S]
+		src.loc << "\blue The decompiler has [amount] units of [S] stored."
+
+
+/obj/item/weapon/matter_decompiler/verb/recompile_matter()
+
+	set name = "Recompile Matter"
+	set desc = "Recompiles stored matter into sheets."
+	set category = "Robot Commands"
+	var/stored = null
+	var/stackamount = null
+
+	var res_type = input("Select a resource to be recompiled:", "Robot", null, null) in src.stored_comms
+	if(res_type)
+		stored = src.stored_comms[res_type]
+		stackamount = round(input("You currently have adequate resources to produce [stored] sheets. How many sheets of [res_type] do you want to produce?") as num)
+	if(stackamount <= 0 || stored <= 0)
+		return
+	if(stackamount > stored) //If the stated amount is greater than biomass actually in the stack
+		stackamount = round(stored)
+
+	//Determine material
+	var/material
+	switch(res_type)
+		if("metal")
+			material = /obj/item/stack/sheet/metal
+			recompile_stacks(stackamount, material, stored, 50, res_type)
```

Comment:
I'm not sure if there is actually a cleaner way to do this without a hassle, but all of these recompile_stacks calls should really just be 1 at the bottom.